modules/custom/openy_upgrade_tool/src/Plugin/Action/ApplyCurrentActiveVersionAction.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Drupal\openy_upgrade_tool\Plugin\Action;
+
+use Drupal\Core\Action\Plugin\Action\EntityActionBase;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Apply current active version for openy_upgrade_log entity.
+ *
+ * @Action(
+ *   id = "apply_active_version",
+ *   action_label = @Translation("Apply current active version"),
+ *   type = "openy_upgrade_log"
+ * )
+ */
+class ApplyCurrentActiveVersionAction extends EntityActionBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute($entity = NULL) {
+    $entity->applyCurrentActiveVersion();
```

Comment:
what if ```$entity == NULL``` as described in method definition? Is it possible to add some sort of error handling to keep safer code?

Reply:
This option is not possible. Also, there is no error handling in similar code in the core. See https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Action%21Plugin%21Action%21SaveAction.php/function/SaveAction%3A%3Aexecute/8.6.x